### PR TITLE
Added provision for resuming training from a checkpoint in case the training is interrupted

### PR DESCRIPTION
**Pull request by sadimanna**
Added provision for resuming training from a checkpoint in case the training is interrupted.

### CODE REVIEW

File: benchmarks/imagenet/resnet50/knn_eval.py [hunk 1]
```diff
@@ -67,6 +67,7 @@ def knn_eval(
     classifier = KNNClassifier(
         model=model,
         num_classes=num_classes,
+        knn_k=20,
```
Comment (guarin) [inline benchmarks/imagenet/resnet50/knn_eval.py:70]: I believe this is unrelated to the PR, could you remove it again?

**Review by guarin — DISMISSED** [review]
Thanks a lot for the PR, this looks great!

I left some comment to change some small things but otherwise this looks good to merge. Let me know if you have time to make the changes, otherwise I can do it later.

File: benchmarks/imagenet/resnet50/knn_eval.py (hunk 1 above)
Comment (sadimanna) [reply benchmarks/imagenet/resnet50/knn_eval.py]: I removed it

File: benchmarks/imagenet/resnet50/main.py [hunk 2]
```diff
@@ -39,6 +39,7 @@
 parser.add_argument("--accelerator", type=str, default="gpu")
 parser.add_argument("--devices", type=int, default=1)
 parser.add_argument("--precision", type=str, default="16-mixed")
+parser.add_argument("--ckpt_path", type=Path, default=None)
```
[the remark this answers is not in the document]
Comment (sadimanna) [reply benchmarks/imagenet/resnet50/main.py]: Made this change

File: benchmarks/imagenet/resnet50/main.py [hunk 3]
```diff
@@ -171,6 +176,7 @@ def pretrain(
     accelerator: str,
     devices: int,
     precision: str,
+    ckpt_path: str = None,
```
[the remark this answers is not in the document]
Comment (sadimanna) [reply benchmarks/imagenet/resnet50/main.py]: I changed this to `Union[Path, None]`

**Comment by sadimanna** [comment]
@guarin  I made the changes as requested.

File: benchmarks/imagenet/resnet50/main.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (guarin) [reply benchmarks/imagenet/resnet50/main.py]: Ah yes good catch!

**Comment by guarin** [comment]
Thanks for the changes! I'll reformat and merge :)

**Review by philippmwirth — APPROVED** [review]
